### ISSUE TITLE
SVGArcApproximator::approximate() doesn't work properly

### ISSUE THREAD

**Issue by captain-igloo** [meyfa/php-svg#77]
I haven't delved too deeply into the maths but the results returned are sometimes incorrect.  For example:
```
$approx = new SVGArcApproximator();
$result = $approx->approximate([10, 10], [20, 10], true, false, 10, 10, 0);
```
The large arc flag is true but it returns a small arc.  I have compared your code with a couple of other implementations. Have a look at [this line](https://github.com/meyfa/php-svg/blob/master/src/Rasterization/Path/SVGArcApproximator.php#L153) and line 155.  $angleDelta should be adjusted, not $angleStart.
Also vectorAngle() returns a positive number so line 155 is never executed (?)
And is fmod() at line 146 needed ?

**Comment by meyfa** [comment]
You are right, the example you gave is indeed computed incorrectly. Investigating.

**Comment by meyfa** [comment]
I hope to have fixed it in 97ff139d483c762259447a814217b14c2af2b5fa. At least your example works for me now, as did some others I tried. Could you perhaps verify this as well, using the newest commit?

**Comment by captain-igloo** [comment]
I have tried a few examples and it seems to be working now.   Thanks.
A small thing - how about adding a test for $sweep = true.  I see line 142 is not covered by your tests currently.
Also I asked before about fmod() line 131.  Since vectorAngle2() returns a number between -pi and pi (?) the fmod shouldn't be needed.
Lastly vectorAngle2() will divide by zero if $vec1x and $vec1y are both zero or if $vec2x and $vec2y are both zero.

**Comment by meyfa** [comment]
I added tests for all 4 combinations of LargeArc and Sweep flags (c9c224478f0e46ac67514831f05c0e03fb232d72). Also, you are right about the fmod call not being needed, I have since removed it (89bc58e393777085b8b4d869933197e55336482b).

Moreover, I considered the division by zero in `vectorAngle` and `vectorAngle2` and came to the conclusion that the formulas provided by W3C are made to prevent this from ever becoming an issue. Entering the formulas into a computer algebra system and solving for inputs that would yield a norm of 0 gave no results. So unless I mistyped something, it won't be a problem.

The only thing that remains, then, is to implement radius scaling in case the ellipse would be too small to cover the endpoints. The algorithm is described in the W3C document as well and I will work on it now.
